src/Xamarin.Legacy.Sdk/Sdk/Sdk.props

```diff
@@ -3,5 +3,21 @@
     <!-- A public property that could be used to detect this SDK -->
     <XamarinLegacySdk>true</XamarinLegacySdk>
   </PropertyGroup>
+
+  <!--
+    HACK: declare a _GetRestoreSettingsPerFramework target that some versions of NuGet expect to exist.
+    This target should get replaced by NuGet.targets if needed:
+    https://github.com/NuGet/NuGet.Client/blob/914cacea77660b8758014912ee127c82469ac1d9/src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets#L635-L651
+  -->
+  <Target Name="_GetRestoreSettingsPerFramework" Returns="@(_RestoreSettingsPerFramework)">
+    <ItemGroup>
+      <_RestoreSettingsPerFramework Include="$([System.Guid]::NewGuid())">
+        <RestoreAdditionalProjectSources>$(RestoreAdditionalProjectSources)</RestoreAdditionalProjectSources>
+        <RestoreAdditionalProjectFallbackFolders>$(RestoreAdditionalProjectFallbackFolders)</RestoreAdditionalProjectFallbackFolders>
+        <RestoreAdditionalProjectFallbackFoldersExcludes>$(RestoreAdditionalProjectFallbackFoldersExcludes)</RestoreAdditionalProjectFallbackFoldersExcludes>
+      </_RestoreSettingsPerFramework>
+    </ItemGroup>
+  </Target>
```

Comment:
@mfkl adding this solved one issue, were you getting errors about this target missing?

After adding this, I hit a new missing target, which is too large to replicate in this project:

https://github.com/NuGet/NuGet.Client/blob/914cacea77660b8758014912ee127c82469ac1d9/src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets#L856-L863

I think these were very late changes to improve no-op restore performance in NuGet in .NET 6.

Need to think on this one...

Reply:
Ok remembering now... This needs to be updated for VS 2022:

https://github.com/xamarin/Xamarin.Legacy.Sdk/blob/4ba5930273eb5bd64b142cfdc574626c68ad41ab/src/Xamarin.Legacy.Sdk/Sdk/Xamarin.VsInstallRoot.targets#L6-L7